code/datums/materials/basemats.dm

```diff
@@ -258,6 +315,43 @@ Unless you know what you're doing, only use the first three numbers. They're in
 	tradable = TRUE
 	tradable_base_quantity = MATERIAL_QUANTITY_EXOTIC
 	texture_layer_icon_state = "shine"
+	fish_weight_modifier = 1.3
+	material_fish_extra_chance = 15
+	fishing_difficulty_modifier = -5
+	fishing_cast_range = 5 //space-bending scifi magic
+	fishing_experience_multiplier = 0.85
+	fishing_completion_speed = 1.1
+	fishing_bait_speed_mult = 1.2
+	fishing_deceleration_mult = 0.9
+	fishing_bounciness_mult = 1.1
+
+/datum/material/bluespace/on_main_applied(atom/source, mat_amount, multiplier)
+	. = ..()
+	if(istype(source, /obj/item/fishing_rod))
+		RegisterSignal(source, COMSIG_ROD_BEGIN_FISHING, PROC_REF(on_begin_fishing))
+
+/datum/material/bluespace/proc/on_begin_fishing(obj/item/fishing_rod/rod, datum/fishing_challenge/challenge, datum/component/fishing_spot/comp)
+	SIGNAL_HANDLER
+	if(prob(67))
+		return
+	comp.fish_source.UnregisterSignal(challenge, list(
+		COMSIG_FISHING_CHALLENGE_ROLL_REWARD,
+		COMSIG_FISHING_CHALLENGE_GET_DIFFICULTY,
+	))
+	var/datum/fish_source/new_source
+	var/list/elegible_fish_sources = GLOB.preset_fish_sources.Copy()
+	for(var/source_type in elegible_fish_sources)
+		var/datum/fish_source/source = elegible_fish_sources[source_type]
+		if(source.fish_source_flags & FISH_SOURCE_FLAG_NO_BLUESPACE_ROD)
+			elegible_fish_sources -= source_type
+	new_source = elegible_fish_sources[pick(elegible_fish_sources)]
+	new_source.RegisterSignal(challenge, COMSIG_FISHING_CHALLENGE_ROLL_REWARD, TYPE_PROC_REF(/datum/fish_source, roll_reward_minigame))
+	new_source.RegisterSignal(challenge, COMSIG_FISHING_CHALLENGE_GET_DIFFICULTY, TYPE_PROC_REF(/datum/fish_source, calculate_difficulty_minigame))
```

Comment:
Material shouldn't be handling source's comsigs, please do something about this, maybe add some way to override the source instead?